src/supautils.c

```diff
@@ -814,15 +797,34 @@ static void supautils_hook(PROCESS_UTILITY_PARAMS) {
           bool already_switched_to_superuser = false;
           const Oid current_user_id = GetUserId();
 
+          CreateEventTrigStmt *stmt = (CreateEventTrigStmt *)utility_stmt;
+          const char *current_role_name = GetUserNameFromId(current_user_id, false);
+
+          bool current_user_is_super = superuser_arg(current_user_id);
+          Oid  function_owner = get_function_owner((func_owner_search){FO_SEARCH_NAME, {stmt->funcname}});
+          bool function_is_owned_by_super = superuser_arg(function_owner);
+
+          if(!current_user_is_super && function_is_owned_by_super){
+            ereport(ERROR, (
+              errmsg("supautils requires a non-superuser event trigger to execute a non-superuser owned function")
```

Comment:
Nit: users shouldn't need to know what "supautils" is
```suggestion
              errmsg("Non-superuser owned event trigger must execute a non-superuser owned function")
```

Reply:
Since it deviates from PostgreSQL standard behavior, I thought it'd be best to be clear as to what component is causing this.

But not adamant on this, so I'll just remove it.